File: internal/content/testdata/bench/typescript/coin-change.ts
function coin_change(coins: number[], amount: number): number {
    const INF = amount + 1;
    const dp: number[] = new Array(amount + 1).fill(INF);
    dp[0] = 0;
    for (let a = 1; a <= amount; a++) {
        for (const c of coins) {
            if (c <= a) {
                dp[a] = Math.min(dp[a], dp[a - c] + 1);
            }
        }
    }
    return dp[amount] === INF ? -1 : dp[amount];
}

File: internal/content/testdata/bench/java/coin-change.java
import java.util.Arrays;

class Solution {
    public long coin_change(long[] coins, long amount) {
        int amt = (int) amount;
        long INF = amt + 1;
        long[] dp = new long[amt + 1];
        Arrays.fill(dp, INF);
        dp[0] = 0;
        for (int a = 1; a <= amt; a++) {
            for (long c : coins) {
                if (c <= a) {
                    dp[a] = Math.min(dp[a], dp[(int)(a - c)] + 1);
                }
            }
        }
        return dp[amt] != INF ? dp[amt] : -1;
    }
}
